src/main/java/it/ipzs/fedauthority/oidclib/ClaimOptions.java

```diff
@@ -0,0 +1,144 @@
+package it.ipzs.fedauthority.oidclib;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import org.json.JSONObject;
+
+import it.ipzs.fedauthority.oidclib.exception.ConfigException;
+import it.ipzs.fedauthority.oidclib.exception.OIDCException;
+import it.ipzs.fedauthority.oidclib.schemas.ClaimItem;
+import it.ipzs.fedauthority.oidclib.schemas.ClaimSection;
+
+public class ClaimOptions {
+
+	private Map<ClaimSection, Set<OptionItem>> itemsMap = new HashMap<>();
+
+	public void addSectionItem(
+			ClaimSection section, ClaimItem claimItem, Boolean essential)
+		throws OIDCException {
+
+		Set<OptionItem> items = itemsMap.get(section);
+
+		if (items == null) {
+			items = new HashSet<>();
+			itemsMap.put(section, items);
+		}
+
+		items.add(OptionItem.of(claimItem, essential));
+	}
+
+	public Set<OptionItem> getSectionItems(ClaimSection section) {
+		Set<OptionItem> items = itemsMap.get(section);
+
+		if (items != null) {
+			return Collections.unmodifiableSet(items);
+		}
+
+		return Collections.emptySet();
+	}
+
+	public boolean hasEssentialItem(String value) {
+		for (OptionItem item : getSectionItems(ClaimSection.ID_TOKEN)) {
+			if (item.isEssential() && item.matchClaimItem(value)) {
+				return true;
+			}
+		}
+
+		for (OptionItem item : getSectionItems(ClaimSection.USER_INFO)) {
+			if (item.isEssential() && item.matchClaimItem(value)) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	public boolean isEmpty() {
+		return itemsMap.isEmpty();
+	}
+
+	public JSONObject toJSON() {
+		JSONObject idToken = new JSONObject();
```

Comment:
id token is something out of scope of federation.

claims and id token must be taken out.